### PR TITLE
preserve the instance shutdown state, and if its bad, don't clean up the test directory

### PR DESCRIPTION
**Pull request by dothebart**
### Scope & Purpose

*(Can you describe what functional change your PR is trying to effect?)*
- fixes testing behviour in shutdown error cases where no test failures occured before.

### Testing & Verification
This functionality comes into account when the shutdown of actual tests fails.

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4651/

File: js/client/modules/@arangodb/testsuites/permissions_server.js [hunk 1]
```diff
@@ -84,22 +84,35 @@ function permissions_server(options) {
...
       } catch (ex) {
+        let shutdownStatus = pu.shutdownInstance(instanceInfo, clonedOpts, false);                                     // stop
         results[testFile] = {
           status: false,
-          messages: 'Warmup of system failed: ' + ex
+          messages: 'Warmup of system failed: ' + ex,
+          shutdown: shutdownStatus
         };
-        pu.shutdownInstance(instanceInfo, clonedOpts, false);                                     // stop
+        results['shutdown'] &= shutdownStatus;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/permissions_server.js]: Please do not use a bitwise operator for a logical operation just for verbosity.
```suggestion
        results['shutdown'] = results['shutdown'] && shutdownStatus;
```

File: js/client/modules/@arangodb/testsuites/permissions_server.js [hunk 2]
```diff
@@ -84,22 +84,35 @@ function permissions_server(options) {
...
+        shutdownStatus = pu.shutdownInstance(instanceInfo, clonedOpts, false);
+      }
+      else {
+        results[testFile] = {
+          status: false,
+          message: "failed to stop instance",
+          shutdown: false
+        };
+      }
+      results['shutdown'] &= shutdownStatus;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/permissions_server.js]: See above.
```suggestion
      results['shutdown'] = results['shutdown'] && shutdownStatus;
```

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 3]
```diff
@@ -82,6 +82,7 @@ function shellReplication (options) {
 
 function replicationFuzz (options) {
   let testCases = tu.scanTestPaths(testPaths.replication_fuzz);
+  let sthutdownStatus = true;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]: typo:
```suggestion
  let shutdownStatus = true;
```

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 4]
```diff
@@ -122,7 +123,7 @@ function replicationFuzz (options) {
                        customInstanceInfos,
                        startStopHandlers) {
       if (pu.arangod.check.instanceAlive(customInstanceInfos.postStart.instanceInfo, options)) {
-        pu.shutdownInstance(customInstanceInfos.postStart.instanceInfo, options);
+        sthutdownStatus = pu.shutdownInstance(customInstanceInfos.postStart.instanceInfo, options);
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]:
```suggestion
        shutdownStatus = pu.shutdownInstance(customInstanceInfos.postStart.instanceInfo, options);
```

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 5]
```diff
@@ -140,8 +141,10 @@ function replicationFuzz (options) {
 
   };
 
-  return tu.performTests(options, testCases, 'replication_fuzz', tu.runInArangosh,
-                         {"rocksdb.wal-file-timeout-initial": "7200"}, startStopHandlers);
+  let results = tu.performTests(options, testCases, 'replication_fuzz', tu.runInArangosh,
+                                {"rocksdb.wal-file-timeout-initial": "7200"}, startStopHandlers);
+  results['shutdown'] &= sthutdownStatus;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]:
```suggestion
  results['shutdown'] = results['shutdown'] && shutdownStatus;
```

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 6]
```diff
@@ -150,6 +153,7 @@ function replicationFuzz (options) {
 
 function replicationRandom (options) {
   let testCases = tu.scanTestPaths(testPaths.replication_random);
+  let sthutdownStatus = true;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]: same typo here and below

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 7]
```diff
@@ -218,6 +224,7 @@ function replicationRandom (options) {
 
 function replicationAql (options) {
   let testCases = tu.scanTestPaths(testPaths.replication_aql);
+  let sthutdownStatus = true;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]: same typo here and below

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 8]
```diff
@@ -287,7 +296,8 @@ function replicationAql (options) {
 var _replicationOngoing = function(path) {
   this.func = function replicationOngoing (options) {
     let testCases = tu.scanTestPaths(testPaths[path]);
-  
+    let sthutdownStatus = true;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]: same typo here and below

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 9]
```diff
@@ -277,7 +284,9 @@ function replicationAql (options) {
 
   };
 
-  return tu.performTests(options, testCases, 'replication_aql', tu.runInArangosh, {}, startStopHandlers);
+  let results = tu.performTests(options, testCases, 'replication_aql', tu.runInArangosh, {}, startStopHandlers);
+  results['shutdown'] &= sthutdownStatus;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]: as above

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 10]
```diff
@@ -363,6 +375,7 @@ const replicationOngoingFrompresent32 = (new _replicationOngoing('replication_on
 
 function replicationStatic (options) {
   let testCases = tu.scanTestPaths(testPaths.replication_static);
+  let sthutdownStatus = true;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]: same typo here and below

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 11]
```diff
@@ -455,6 +470,7 @@ function replicationStatic (options) {
 
 function replicationSync (options) {
   let testCases = tu.scanTestPaths(testPaths.replication_sync);
+  let sthutdownStatus = true;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]: same typo here and below

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 12]
```diff
@@ -209,7 +213,9 @@ function replicationRandom (options) {
 
   };
 
-  return tu.performTests(options, testCases, 'replication_random', tu.runInArangosh, {}, startStopHandlers);
+  let results = tu.performTests(options, testCases, 'replication_random', tu.runInArangosh, {}, startStopHandlers);
+  results['shutdown'] &= sthutdownStatus;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]: as above

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 13]
```diff
@@ -346,7 +356,9 @@ var _replicationOngoing = function(path) {
   
     };
   
-    return tu.performTests(options, testCases, path, tu.runInArangosh, {}, startStopHandlers);
+    let results = tu.performTests(options, testCases, path, tu.runInArangosh, {}, startStopHandlers);
+    results['shutdown'] &= sthutdownStatus;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]: as above

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 14]
```diff
@@ -447,6 +460,8 @@ function replicationStatic (options) {
       'server.authentication': 'true'
     },
     startStopHandlers);
+  results['shutdown'] &= sthutdownStatus;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]: as above

File: js/client/modules/@arangodb/testsuites/replication.js [hunk 15]
```diff
@@ -529,7 +545,9 @@ function replicationSync (options) {
     }
   };
 
-  return tu.performTests(options, testCases, 'replication_sync', tu.runInArangosh, {}, startStopHandlers);
+  let results = tu.performTests(options, testCases, 'replication_sync', tu.runInArangosh, {}, startStopHandlers);
+  results['shutdown'] &= sthutdownStatus;
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/replication.js]: as above

File: js/client/modules/@arangodb/testsuites/single.js [hunk 16]
```diff
@@ -152,7 +152,7 @@ function singleServer (options) {
     options.cleanup = false;
   }
 
-  pu.shutdownInstance(instanceInfo, options);
+  result['shutdon'] = pu.shutdownInstance(instanceInfo, options);
```
Comment (goedderz) [inline js/client/modules/@arangodb/testsuites/single.js]: typo:
```suggestion
  result['shutdown'] = pu.shutdownInstance(instanceInfo, options);
```

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/4653/

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/6211/

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/6220/

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/6222/

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr-linux/6228/EDITION=community,STORAGE_ENGINE=rocksdb,TEST_SUITE=single,limit=linux&&test/

**Pull request merged**